RecoCTPPS/TotemRPLocal/src/CTPPSDiamondRecHitProducerAlgorithm.cc

```diff
@@ -34,25 +45,49 @@ CTPPSDiamondRecHitProducerAlgorithm::build( const CTPPSGeometry* geom, const edm
                 y_width = 2.0 * det->params().at( 1 ),
                 z_width = 2.0 * det->params().at( 2 );
 
+    // retrieve the timing calibration part for this channel
+    const int sector = detid.arm(), station = detid.station(), plane = detid.plane(), channel = detid.channel();
+    const auto& ch_params = calib_.parameters( sector, station, plane, channel );
+    // offset + time precision set to 0 if not found
+    const double ch_t_offset = calib_.timeOffset( sector, station, plane, channel );
+    const double ch_t_precis = calib_.timePrecision( sector, station, plane, channel );
+
     edm::DetSet<CTPPSDiamondRecHit>& rec_hits = output.find_or_insert( detid );
 
     for ( const auto& digi : vec ) {
-      if ( digi.getLeadingEdge() == 0 && digi.getTrailingEdge() == 0 ) continue;
-
-      const int t = digi.getLeadingEdge();
-      const int t0 = t % 1024;
-      const int time_slice = ( t != 0 ) ? t / 1024 : CTPPSDiamondRecHit::TIMESLICE_WITHOUT_LEADING;
-
-      int tot = 0;
-      if ( t != 0 && digi.getTrailingEdge() != 0 ) tot = ( (int)digi.getTrailingEdge() ) - t;
-
-      rec_hits.push_back( CTPPSDiamondRecHit( x_pos, x_width, y_pos, y_width, z_pos, z_width, // spatial information
-                                              ( t0 * ts_to_ns_ ),
-                                              ( tot * ts_to_ns_),
-                                              0., // time precision
-                                              time_slice,
-                                              digi.getHPTDCErrorFlags(),
-                                              digi.getMultipleHit() ) );
+      const int t_lead = digi.getLeadingEdge(), t_trail = digi.getTrailingEdge();
+      // skip invalid digis
+      if ( t_lead == 0 && t_trail == 0 )
+        continue;
+
+      double tot = -1., ch_t_twc = 0.;
+      if ( t_lead != 0 && t_trail != 0 ) {
+        tot = ( t_trail-t_lead )*ts_to_ns_; // in ns
+        if ( calib_fct_ ) {
+          // compute the time-walk correction
+          ch_t_twc = calib_fct_->evaluate( std::vector<double>{ tot }, ch_params );
+          if ( std::isnan( ch_t_twc ) )
```

Comment:
static analyzer flagged this
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26327/34130/llvm-analysis/report-2166ef.html#EndPath
```Please use edm::isNotFinite from 'FWCore/Utilities/interface/isNotFinite.h'```

Reply:
Good catch! However the include path seems outdated... Attempted to fix in #26435.